anitya/lib/__init__.py

```diff
@@ -262,3 +262,62 @@ def map_project(
     )
 
     return pkg
+
+
+def flag_project(session, project, reason, user_mail):
+    """ Flag a project in the database.
+
+    """
+
+    flag = anitya.lib.model.ProjectFlag(
+        user=user_mail,
+        project=project,
+        reason=reason)
+
+    session.add(flag)
+
+    try:
+        session.flush()
+    except SQLAlchemyError as err:
+        log.exception(err)
+        session.rollback()
+        raise anitya.lib.exceptions.AnityaException(
+            'Could not flag this project.')
+
+    anitya.log(
+        session,
+        project=project,
+        topic='project.flag',
+        message=dict(
+            agent=user_mail,
+            project=project.name,
+            reason=reason,
+        )
+    )
+    session.commit()
+    return project
+
+def set_flag_state(session, flag, state, user_mail):
+    """ Change the state of a ProjectFlag in the database.
+
+    """
+
+    flag.state = state
```

Comment:
Ahh, I see what you mean, I think. Because someone could just keep hitting the URL that sets the flag to the same state over and over again, right? I'll add a check for that now.